LearningMissionSimulation/PlaygroundVahe.cs

```diff
@@ -156,17 +247,177 @@ List<Module> GetModuleList()
+                {
+                    Guid moduleId = classroom.Module.Id;
+                    List<Instructor> instructorList;
+                    if (ModuleInstructorListDictionary.ContainsKey(moduleId))
+                    {
+                        ModuleInstructorListDictionary.TryGetValue(moduleId, out instructorList);
+                        classroom.Head = instructorList[AttributeGenerator.random.Next(0, instructorList.Count)];
+                    }
+                }
+            }
+            else if (ActiveInstructorList.Count == 0)
+            {
+                ReportError(ActiveInstructorList.ToString(), action);
+                if (ClassroomDictionary.Count == 0)
+                {
+                    ReportError(ClassroomDictionary.ToString(), action);
+                }
+            }
+            ReportSummary(action, 1, "classroom");
+            ReportFooter(action);
+        }
+
+        void AddInstructorToModuleList(Instructor instructor)
+        {
+            List<Module> moduleList = instructor.ModuleList;
+            foreach (var module in moduleList)
+            {
+                List<Instructor> instructorList;
+                if (ModuleInstructorListDictionary.ContainsKey(module.Id))
+                {
+                    ModuleInstructorListDictionary.TryGetValue(module.Id, out instructorList);
+                    instructorList.Add(instructor);
+                }
+                else
+                {
+                    instructorList = new List<Instructor>() { instructor };
+                    ModuleInstructorListDictionary.Add(module.Id, instructorList);
+                }
+            }
         }
 
         public void RegisterStudentsForClasses()
         {
-            throw new NotImplementedException();
+            string action = "Register Students For Classes";
+            int itemListCount = 0;
+            ReportHeader(action);
+
+            if (ClassroomDictionary.Count > 0)
+            {
+                foreach (var classroom in ClassroomDictionary.Values)
+                {
+                    if (classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                        itemListCount = classroom.ItemList.Count;
+                    }
+                }
+            }
+            else
+            {
+                ReportError(ClassroomDictionary.ToString(), action);
+            }
+
+            ReportSummary(action, itemListCount, "student");
+            ReportFooter(action);
+        }
+
+        void UpdateStudentList(Classroom classroom)
+        {
+            string action = "Register Students For Classes";
+            uint itemListCount = (uint)AttributeGenerator.random.Next(classroom.MaximumCapacity - classroom.ItemList.Count + 1);
+
+            for (int i = 1; i <= itemListCount; i++)
+            {
+                foreach (Student student in ActiveStudentList)
```

Comment:
Your **foreach** loop does not put any restrictions on the number of students registered for the class. If students in the active students list, did not take the class before, you allow to register them ALL to the class. This is wrong.